packages/client/src/task-queue-client.ts

```diff
@@ -0,0 +1,297 @@
+   *  reachable, the map value will be an empty array.
+   */
+  taskQueueReachability: Record<string, ReachabilityTypeResponse[]>;
+}
+
+function reachabilityTypeToProto(type: ReachabilityType | undefined | null): temporal.api.enums.v1.TaskReachability {
+  switch (type) {
+    case null:
+    case undefined:
+      return temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_UNSPECIFIED;
+    case 'NewWorkflows':
+      return temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_NEW_WORKFLOWS;
+    case 'ExistingWorkflows':
+      return temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_EXISTING_WORKFLOWS;
+    case 'OpenWorkflows':
+      return temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_OPEN_WORKFLOWS;
+    case 'ClosedWorkflows':
+      return temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_CLOSED_WORKFLOWS;
+    default:
+      assertNever(type);
+  }
+}
+
+export function reachabilityResponseFromProto(resp: GetWorkerTaskReachabilityResponse): ReachabilityResponse {
+  const buildIdReachability: Record<keyof any, BuildIdReachability> = {};
+  for (const bir of resp.buildIdReachability) {
+    const taskQueueReachability: Record<string, ReachabilityTypeResponse[]> = {};
+    if (bir.taskQueueReachability != null) {
+      for (const tqr of bir.taskQueueReachability) {
+        if (tqr.taskQueue == null) {
+          continue;
+        }
+        if (tqr.reachability == null) {
+          taskQueueReachability[tqr.taskQueue] = [];
+          continue;
+        }
+        taskQueueReachability[tqr.taskQueue] = tqr.reachability.map(reachabilityTypeFromProto);
+      }
+    }
+    buildIdReachability[bir.buildId ?? UnversionedBuildId] = { taskQueueReachability };
+  }
+  return {
+    buildIdReachability,
+  };
+}
+
+function reachabilityTypeFromProto(rtype: temporal.api.enums.v1.TaskReachability): ReachabilityTypeResponse {
+  switch (rtype) {
+    case temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_UNSPECIFIED:
+      return 'NotFetched';
+    case temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_NEW_WORKFLOWS:
+      return 'NewWorkflows';
+    case temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_EXISTING_WORKFLOWS:
+      return 'ExistingWorkflows';
+    case temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_OPEN_WORKFLOWS:
+      return 'OpenWorkflows';
+    case temporal.api.enums.v1.TaskReachability.TASK_REACHABILITY_CLOSED_WORKFLOWS:
+      return 'ClosedWorkflows';
+    default:
+      return assertNever(rtype);
+  }
+}
+
+/**
+ * Thrown when one or more Build Ids are not found while using the {@link TaskQueueClient}.
+ *
+ * It could be because:
+ * - Id passed is incorrect
+ * - Build Id has been scavenged by the server.
+ *
+ * @experimental
+ */
+export class BuildIdNotFoundError extends Error {
+  public readonly name: string = 'BuildIdNotFoundError';
+
+  constructor(message: string) {
```

Comment:
If you're not tired from me discovering things every round, you could specify which build id is not found in this exception, could have a `buildId: string` attribute on this class.

Reply:
You can't actually do that, because in the merge operation it could be either of two and the error isn't readily parseable. Mentioned here: https://github.com/temporalio/sdk-typescript/pull/1146#discussion_r1244088033